completions/tmux

```diff
@@ -0,0 +1,134 @@
+# tmux(1) completion
+
+_tmux()
+{
+    local cur prev opts onePrev
+    _init_completion || return
+
+    COMPREPLY=()
+    cur="${COMP_WORDS[COMP_CWORD]}"
+    onePrev="${COMP_WORDS[COMP_CWORD-1]}"
+
+    if [ "$COMP_CWORD" -ge 2 ]; then
+        # Check if we want to list available sessions. We're going to assume
+        # this is defined as: [ls | list-session] -t .
+        local windowCommands currentSessions
+        windowCommands=("ls", "list-sessions")
+
+        prev="${COMP_WORDS[COMP_CWORD-2]}"
+        if [ "$prev" = "attach" ] || [ "$prev" = "attach-session" ] ; then
+            if [ "$onePrev" = "-t" ] ; then
+            # Get a list of all session names.
+            # We're assuming this output is in the form:
+            # 'name: more info'
+            # We'll get the name by using cut--use ":" as a delimiter, and
+            # print the first column. If there are no sessions, we expect the
+            # following message on stderr:
+            # 'failed to connect to server: Connection refused'.
+            # In this case we'll return an empty list.
+            # NB: Since the failed to connect message is displayed on stderr,
+            # we need to pipe both stdout and stderr to cut. Therefore we are
+            # piping with |& rather than just |.
+            currentSessions=$(tmux ls |& cut -d : -f 1)
+            if [ "${currentSessions[0]}" = "failed to connect to server" ]; \
+            then
+                # We don't want to display any options, so clear the array.
+                currentSessions=$( )
+            fi
+            COMPREPLY=($(compgen -W "${currentSessions}" -- ${cur}))
+            return
+            fi
+        fi
+    fi
+
+    opts=" \
+    attach-session \
```

Comment:
Not a big fan of this long hardcoded list in the first place. Can't it be generated from somewhere?

If not, please indent the options, and the backslashes are unnecessary I think.